astropy/modeling/rotations.py

```diff
@@ -187,9 +187,6 @@ def evaluate(cls, x, y, angle):
         single coordinate pair.
         """
 
-        if x.shape != y.shape:
-            raise ValueError("Expected input arrays to have the same shape")
```

Comment:
Without this workaround `Rotation2D` cannot be used in compound models. I'd prefer to do the change here and open a separate issue for the problem. This has no effect on using it as a single model.